### PR TITLE
Add more details to the failure to admit event and condition message

### PR DESCRIPTION
**Pull request by ahg-g**
#### What type of PR is this?

Add the reason for failing to admit a workload in the event and condition message. We can still do more by plumbing reason related to failed to match affinity or taints. 

/kind cleanup

### CODE REVIEW

File: pkg/scheduler/scheduler.go [hunk 1]
```diff
@@ -153,25 +164,20 @@ func (s *Scheduler) nominate(ctx context.Context, workloads []workload.Info, sna
...
-		if !cq.NamespaceSelector.Matches(labels.Set(ns.Labels)) {
-			log.V(2).Info("Workload namespace doesn't match clusterQueue selector")
-			continue
-		}
 		e := entry{Info: w}
-		if !e.assignFlavors(log, snap.ResourceFlavors, cq) {
-			log.V(2).Info("Workload didn't fit in remaining clusterQueue even when borrowing")
-			continue
+		if cq == nil {
+			e.noAdmissionReason = "ClusterQueue not found when calculating workload admission requirements"
```
Comment (alculquicondor) [inline pkg/scheduler/scheduler.go:170]:
```suggestion
			e.noAdmissionReason = "ClusterQueue not found"
```

I think the rest of the string only make sense as part of the log, not as a user-visible message.

File: pkg/scheduler/scheduler.go [hunk 2]
```diff
@@ -153,25 +164,20 @@ func (s *Scheduler) nominate(ctx context.Context, workloads []workload.Info, sna
...
+		} else if err := s.client.Get(ctx, types.NamespacedName{Name: w.Obj.Namespace}, &ns); err != nil {
+			e.noAdmissionReason = err.Error()
```
Comment (alculquicondor) [inline pkg/scheduler/scheduler.go:172]:
```suggestion
			e.noAdmissionReason = fmt.Sprintf("Could not obtain workload namespace: %v", err)
```

File: pkg/scheduler/scheduler.go [hunk 3]
```diff
@@ -153,25 +164,20 @@ func (s *Scheduler) nominate(ctx context.Context, workloads []workload.Info, sna
...
+		} else if !cq.NamespaceSelector.Matches(labels.Set(ns.Labels)) {
+			e.noAdmissionReason = "Workload namespace doesn't match clusterQueue selector"
```
Comment (alculquicondor) [inline pkg/scheduler/scheduler.go:174]:
```suggestion
			e.noAdmissionReason = "Workload namespace doesn't match ClusterQueue selector"
```

File: pkg/scheduler/scheduler.go [hunk 4]
```diff
@@ -153,25 +164,20 @@ func (s *Scheduler) nominate(ctx context.Context, workloads []workload.Info, sna
...
+		} else if !e.assignFlavors(log, snap.ResourceFlavors, cq) {
+			e.noAdmissionReason = "Workload didn't fit remaining quota even when borrowing"
```
Comment (alculquicondor) [inline pkg/scheduler/scheduler.go:176]:
```suggestion
			e.noAdmissionReason = "Workload didn't fit in the remaining quota"
```

some setups might not have cohorts, so borrowing might be confusing.

File: pkg/scheduler/scheduler.go (hunk 2 above)
Comment (ahg-g) [reply pkg/scheduler/scheduler.go:172]: the error message should include context, no?

Comment (alculquicondor) [reply pkg/scheduler/scheduler.go:172]: sure, but the user might be confused on why there is a message for a namespace failure

**Comment by alculquicondor** [comment]
Please fix `vet`

File: pkg/scheduler/scheduler.go (hunk 2 above)
Comment (ahg-g) [reply pkg/scheduler/scheduler.go]: not sure if saying "couldn't obtain workload namespace" removes that confusion, but I updated the message as suggested :)

**Comment by denkensk** [comment]
/lgtm

Maybe you need to squash it.

**Pull request merged**